src/DeviceInput/Implementations/webDeviceInputSystem.ts

```diff
@@ -415,12 +420,18 @@ export class WebDeviceInputSystem implements IDeviceInputSystem {
                 this._addPointerDevice(deviceType, deviceSlot, evt.clientX, evt.clientY);
             }
 
+            if (deviceType === DeviceType.Mouse && this._mouseId === -1) {
+                this._mouseId = evt.pointerId;
+            }
+
             const pointer = this._inputs[deviceType][deviceSlot];
             if (pointer) {
                 const previousHorizontal = pointer[PointerInput.Horizontal];
                 const previousVertical = pointer[PointerInput.Vertical];
                 const previousButton = pointer[evt.button + 2];
 
+                this._elementToAttachTo.setPointerCapture(evt.pointerId);

```

Comment:
Was it the case before? if no what is the rationale behind this choice?

Reply:
I believe this is because we want to capture pointerup events even if the pointer leaves the canvas before the user releases the button. Since the element captures the pointer, it will receive the pointerup event no matter where the pointer is on screen. https://developer.mozilla.org/en-US/docs/Web/API/Element/setPointerCapture

This is relevant to #10353  because the user may move their mouse outside the texture inspector window before releasing the pan button. The texture inspector need to catch this pointerup event and stop panning.

@PolygonalSun do I understand correctly?

Reply:
Yup, that's correct.  Initially, we were just listening to the window for pointerup events and handling it that way but that method doesn't work with situations that have multiple windows like the Texture Inspector.  This approach gives us identical behavior with single windows but also fixes pointerup listening for the pop out window.

Reply:
Ok make sure it works when inside NME preview window, Playground, etc...

Reply:
There is an edge case where this doesn't work correctly. If you hold down two different buttons and let go of only one of them, the pointer capture will be released. Seems like we need to ref count for this to be totally correct.

Reply:
I wasn't able to repro this edge case result.  When I tried tracking this what I got was
FIRST BUTTON DOWN (pointerdown fired, capture set)
SECOND BUTTON DOWN (pointermove fired, capture still set)
ANY BUTTON UP (pointermove fired, capture still set)
OTHER BUTTON UP (pointerup fired, capture released)

The pointercapturelost event only fired after released in the pointerup (after the second button was released).  I've tried this with both the PG and Texture Inspector.  Is there a step that I might be missing to repro this?

Reply:
Sorry, I didn't actually try this code change. I mentioned this because I tried the [MDN sample](https://developer.mozilla.org/en-US/docs/Web/API/Element/setPointerCapture) and it didn't work right for this. Maybe this code works and I don't understand it well enough to see it.

Reply:
Just as an update to what this changes:
- NME: No change
- GUI Editor: No change
- Sandbox: PointerUp working in Texture Inspector window; Objects can be dragged off screen (drag coords can be read off window)
- Playground: Objects can be dragged off screen (drag coords can be read off window); works with Gizmos
While this is a slight change to behavior, I don't see this as a negative change.  The original behavior was that if the cursor left the screen while a button was pressed, the dragging behavior would stop but the object being dragged would "snap" to the coordinates of the cursor when it returned, assuming the button was never released.

Reply:
This is pretty neat actually :D